### DISCUSSION TITLE
Flight lines visible in gridded airborne geophysics

### DISCUSSION THREAD

**Discussion by CavemanEx** [fatiando/community#126] [Q&A] [answered]
Hi all,
This might be a pretty basic question, but I've been playing around with gridding some raw airborne EM data and the results clearly show elevated values over the flight lines.

The program is working great and I've been able to chain verde operations in order to get perfectly usable rasters.

One minor problem I've been having is that the program has a hard time interpolating between the flight lines and this gives the final result a striped look:
<img width="449" alt="image" src="https://github.com/fatiando/community/assets/62568183/5c181d0b-ed7a-45a0-8f84-e7570b14921c">

This is particularly noticeable above and below strong anomalies: 
<img width="280" alt="image" src="https://github.com/fatiando/community/assets/62568183/80f28135-ec92-448d-a3ad-9264fd0afd0e">

I assume there are more settings to play with in Verde.spline? I tried using the spline parameter "damping" which helped even out some of the noise, but the striped flight patterns didn't fully resolve. 

Any help you could give for this would be awesome, Thanks!

Harley

**Comment by leouieda** [comment, marked as answer]
Hi @CavemanEx, welcome to the forum! This is a common problem with interpolation of airborne geophysics. The main issue is that data are oversampled along flight lines and there no information between the lines. If you interpolate with a small grid spacing to take advantage of the large sampling along lines, the interpolator has to make up data between lines. Most interpolators will try to smooth and decay away from actual data. This is what creates the patterns along flight lines (the many points between lines tend to decay until they get close to another line).

There aren't really good solutions to this problem. Some things that can be done:

1. Use directional filters to artificially remove the stripes. This is what some people do but I don't really think it's a good way to do this since it can remove actual signals in the flight direction.
2. Smooth the grid with a low-pass filter. I also don't recommend this since it's wasteful to create a fine grid and then smooth it.
3. Grid with a larger spacing that is more compatible with the flight line separation. Yes, you will lose some information along flight paths but this is inevitable if you want a grid in the end. This is my main recommendation. 

If it helps, there is this example of how I made a grid of airborne magnetic data for Ensaio: https://www.fatiando.org/ensaio/latest/gallery/lightning-creek-magnetic.html#sphx-glr-gallery-lightning-creek-magnetic-py (this shows the grid and there is a link to the code that generated it). It uses equivalent sources, which are very similar to the gridding in Verde.

**Comment by CavemanEx** [comment]
Awesome, thanks for the reply!
That makes sense, I've even seen flight lines visible in final deliverables from geophysics companies. 
As far as optimizing the grid size, is there a rule of thumb where your blocks should be a certain size relative to the distance between flight lines?
Or is it more of a trial and error sort of thing?

This is a great project by the way! I've found it extremely helpful in everything from visualizing and interpreting geophysics to geochemical samples.

**Comment by leouieda** [comment]
Great to know @CavemanEx! Please share any work you did using Fatiando under the Show And Tell category here in the forum: https://github.com/orgs/fatiando/discussions/categories/show-and-tell We love learning what people are up to and it can help us plan the development and justify the effort we put into this.

> As far as optimizing the grid size, is there a rule of thumb where your blocks should be a certain size relative to the distance between flight lines? Or is it more of a trial and error sort of thing?

That is a great question that I don't really have a good answer for. I feel that there must be some relationship but I've never seen anything published on this that was convincing. There might be something in variogram analysis that could provide insight but that's as far as I've gone. If you do find something, please share!
